### PR TITLE
Clean up parse_dsn

### PR DESCRIPTION
**Pull request by rkleemann**
Changs to parse_dsn to make it more readable, and perhaps a little more efficient.

### CODE REVIEW

**Comment by rkleemann** [comment]
Strangely, the Travis check shows a fail, even though every unit in it passed?  Why is that?

**Comment by choroba** [comment]
> every unit in it passed

It didn't: https://travis-ci.org/gooddata/DBD-MariaDB/jobs/492973262

```
DBD::MariaDB::db do failed: Deadlock found when trying to get lock; try restarting transaction at t/41int_min_max.t line 30.
# Looks like you planned 616 tests but ran 20.
# Looks like your test exited with 11 just after 20.
t/41int_min_max.t ....................... Dubious, test returned 11 (wstat 2816, 0xb00)
```

**Comment by choroba** [comment]
It's a known issue we're still investigating. I restarted the job.

**Comment by choroba** [comment]
We don't want to expose the aliases. We support them for historical reasons, we don't plan to add any new aliases, so we prefer to keep them hidden so no one gets tricked into implementing another one.

Also, we don't find `\x5b` more readable than `\[` (and similarly for other characters).

Therefore, there's not much left to merge. We appreciate your effort and hope we haven't scared you off by this refusal.

**Comment by rkleemann** [comment]
Thanks to your comments, I made some changes.  Accept them or reject them as you see appropriate, I will not be scared off.

File: lib/DBD/MariaDB.pm [hunk 1]
```diff
@@ -64,36 +64,30 @@ sub CLONE {
...
-	    if ($var eq 'hostname'  ||  $var eq 'host') {
-		$hash->{'host'} = $val;
-	    } elsif ($var eq 'db'  ||  $var eq 'dbname') {
-		$hash->{'database'} = $val;
-	    } else {
-		$hash->{$var} = $val;
+	if ($part =~ /([^=]*)=(.*)/) {
+	    my ($var, $val) = ($1, $2);
+	    # These are legacy, do not change, and do not add
+	    if ( $var eq 'db' || $var eq 'dbname' ) {
```
Comment (pali) [inline lib/DBD/MariaDB.pm]: Please do not mix tabs and spaces as it just cause problems for readers who have different tabstop settings in editor. If you are rewriting whole function (like this one) then please use 4 spaces (not tabs) for indentation.

Comment (rkleemann) [reply lib/DBD/MariaDB.pm]: I was previously using the GitHub editor, which used whatever indentation was already there.  My most recent changes have updated to be space-only indentations.

**Comment by choroba** [comment]
Look better now. Please, rebase & squash the commits so that the unnecessary changes aren't present at all. Also, the tests should go to a new test file (they aren't related to the `data_sources` method), the commit introducing them should go first (because the test should pass even without the changes in the code) and it should test all the possible code paths, not just two random examples.

**Pull request closed without merging**